### PR TITLE
Feature: add xc_temperature input keyword for finite T functionals

### PR DESCRIPTION
**Pull request by wenfei-li**

### CODE REVIEW

File: docs/input-main.md [hunk 1]
```diff
@@ -272,6 +272,11 @@ This part of variables are used to control general system parameters.
     
 - **Default**: same as UPF file.
 
+#### xc_termperature
```
Comment (caic99) [inline docs/input-main.md]:
```suggestion
#### xc_temperature
```

File: docs/input-main.md [hunk 2]
```diff
@@ -272,6 +272,11 @@ This part of variables are used to control general system parameters.
...
+- **Type**: Double
+- **Description**: specifies temperature when using temperature-dependent XC functionals (KSDT and so on)
+- **Default** : 0.0
```
Comment (caic99) [inline docs/input-main.md:278]: Do we need a unit for this?

File: source/module_xc/xc_functional.cpp [hunk 3]
```diff
@@ -230,7 +230,13 @@ std::vector<xc_func_type> XC_Functional::init_func(const int xc_polarized)
 
 	for(int id : func_id)
 	{
-		if( id == XC_HYB_GGA_XC_PBEH ) // PBE0
+        if( id == XC_LDA_XC_KSDT || id == XC_LDA_XC_CORRKSDT || id == XC_LDA_XC_GDSMFB) //finite temperature XC functionals
+        {
+            add_func ( id );
+            double parameter_finitet[1] = { GlobalV::XC_TEMPERATURE};
+            xc_func_set_ext_params(&funcs.back(), parameter_finitet);
```
Comment (caic99) [inline source/module_xc/xc_functional.cpp]:
```suggestion
        if(id == XC_LDA_XC_KSDT || id == XC_LDA_XC_CORRKSDT || id == XC_LDA_XC_GDSMFB) //finite temperature XC functionals
        {
            add_func(id);
            double parameter_finitet[1] = {GlobalV::XC_TEMPERATURE};
            xc_func_set_ext_params(&funcs.back(), parameter_finitet);
```

**Review by caic99 — DISMISSED** [review]

File: docs/input-main.md (hunk 2 above)
Comment (wenfei-li) [reply docs/input-main.md:278]: > Do we need a unit for this?

indeed. working on it

**Review by Qianruipku — APPROVED** [review]

**Pull request merged**